relay-server/src/actors/project.rs

```diff
@@ -506,21 +498,34 @@ impl Project {
         }
     }
 
+    pub fn merge_rate_limits(&mut self, rate_limits: RateLimits) {
+        self.rate_limits.merge(rate_limits);
+    }
+
     /// Returns a reference to the project state if available.
     pub fn state(&self) -> Option<&ProjectState> {
         self.state.as_deref()
     }
 
+    /// Returns a reference to the project state if available.
+    pub fn state_clone(&self) -> Option<Arc<ProjectState>> {
+        self.state.clone()
+    }
+
     /// Creates the aggregator if it is uninitialized and returns it.
     ///
     /// Returns `None` if the aggregator is permanently disabled, primarily for disabled projects.
-    fn get_or_create_aggregator(
+    pub fn get_or_create_aggregator(
```

Comment:
Ideally, we do not have to expose this. Please rather expose methods corresponding to the former messages, such as `fn merge_buckets` or `fn insert_metrics`.

Reply:
ok, done